src/rgw/rgw_directory.cc

```diff
@@ -0,0 +1,332 @@
+#include <errno.h>
+#include <cpp_redis/cpp_redis>
+#include "rgw_directory.h"
+#include <string>
+#include <iostream>
+#include <sstream>
+#include <algorithm>
+#include <vector>
+#include <list>
+
+void RGWBlockDirectory::findClient(string key, cpp_redis::client *client, int port) {
+  if (client->is_connected()) 
+    return;
+  
+  client->connect("127.0.0.1", port, nullptr);
+  
+  if (!client->is_connected())
+    exit(1);
+}
+
+string RGWBlockDirectory::buildIndex(cache_block *ptr) {
+  return ptr->c_obj.obj_name;
```

Comment:
If we remove the `cache_obj` struct entirely this return will need to be tweaked